carotene: possible fall through

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => recent 3.4( b584c230613705822e75ccae556ad2d642b67a94 ) /master( f9cefc84a )
- Operating System / Platform => Ubuntu 18.04 Aarch64
- Compiler => GCC 7.4.0

##### Detailed description

Warning was raised and I think it's worth fixing it

```
[ 33%] Building CXX object modules/imgproc/CMakeFiles/opencv_imgproc.dir/src/filter.dispatch.cpp.o
In file included from /home/odroid/opencv-fork/build/custom_hal.hpp:5:0,
                 from /home/odroid/opencv-fork/modules/imgproc/src/hal_replacement.hpp:783,
                 from /home/odroid/opencv-fork/modules/imgproc/src/precomp.hpp:54,
                 from /home/odroid/opencv-fork/modules/imgproc/src/filter.dispatch.cpp:43:
/home/odroid/opencv-fork/build/carotene/tegra_hal.hpp: In function ‘int TEGRA_FILTERINIT(cvhalFilter2D**, uchar*, size_t, int, int, int, int, int, int, int, int, double, int, int, bool, bool)’:
/home/odroid/opencv-fork/build/carotene/tegra_hal.hpp:1129:9: warning: this statement may fall through [-Wimplicit-fallthrough=]
         }
         ^
/home/odroid/opencv-fork/build/carotene/tegra_hal.hpp:1130:5: note: here
     default:
     ^~~~~~~
/home/odroid/opencv-fork/build/carotene/tegra_hal.hpp: In function ‘int TEGRA_SEPFILTERINIT(cvhalFilter2D**, int, int, int, uchar*, int, uchar*, int, int, int, double, int)’:
/home/odroid/opencv-fork/build/carotene/tegra_hal.hpp:1245:29: warning: this statement may fall through [-Wimplicit-fallthrough=]
         ctx->kernely_data[2]=((int16_t*)kernely_data)[2];
         ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/odroid/opencv-fork/build/carotene/tegra_hal.hpp:1246:5: note: here
     default:
     ^~~~~~~
```

The code is [here](https://github.com/opencv/opencv/blob/b584c230613705822e75ccae556ad2d642b67a94/3rdparty/carotene/hal/tegra_hal.hpp#L1125-L1130) and [here](https://github.com/opencv/opencv/blob/b584c230613705822e75ccae556ad2d642b67a94/3rdparty/carotene/hal/tegra_hal.hpp#L1239-L1246)

https://github.com/opencv/opencv/blob/b584c230613705822e75ccae556ad2d642b67a94/3rdparty/carotene/hal/tegra_hal.hpp#L1125-L1130

https://github.com/opencv/opencv/blob/b584c230613705822e75ccae556ad2d642b67a94/3rdparty/carotene/hal/tegra_hal.hpp#L1239-L1246

The code is untouched for long time, so it doesn't make big difference, but reading carefully before and after the spotted line, I guess ```break``` is missing

##### Steps to reproduce

Build with GCC 7.4.0 on Aarch64

I'll send a patch later

--- a/3rdparty/carotene/hal/tegra_hal.hpp
+++ b/3rdparty/carotene/hal/tegra_hal.hpp
@@ -1127,6 +1127,7 @@ inline int TEGRA_FILTERINIT(cvhalFilter2D **context, uchar *kernel_data, size_t 
         {
             std::memcpy(ctx->kernel_data + kernel_width * j, kernel_data + kernel_step * j, kernel_width * sizeof(int16_t));
         }
+        break;
     default:
         delete[] ctx->kernel_data;
         delete ctx;
@@ -1243,6 +1244,7 @@ inline int TEGRA_SEPFILTERINIT(cvhalFilter2D **context, int src_type, int dst_ty
         ctx->kernely_data[0]=((int16_t*)kernely_data)[0];
         ctx->kernely_data[1]=((int16_t*)kernely_data)[1];
         ctx->kernely_data[2]=((int16_t*)kernely_data)[2];
+        break;
     default:
         delete ctx;
         return CV_HAL_ERROR_NOT_IMPLEMENTED;

